File: src/ashwini/model/conductivity.jl
include("electrolyte_conc.jl")

# Effective electronic conductivity

"""
Effective electronic conductivity in negative electode
"""
σ₁ₙ = k₁ₙ * (ϵ₁ₙ ^ brugₙ)

"""
Effective electronic conductivity in positive electode
"""
σ₁ₚ = k₁ₚ * (ϵ₁ₚ ^ brugₚ)

# Ionic conductivity

"""
Electrloyte conductivity [Ref45 - 25]
"""
function k₂(c₂)
    1e-4*c₂*(-10.5 + 0.074*T - 6.69e-5*T^2 + 6.68e-4*c₂ - 1.78e-5*c₂*T + 2.8e-8*c₂*T^2 + 4.94e-7*c₂.^2 - 8.86e-10*T*c₂.^2).^2
end

"""
Effective Electrloyte conductivity in negative electrode
"""
function k₂ₙ(params::Vector{Float64})
    k₂(c₂̄ₙ(params)) * (ϵ₂ₙ ^ brugₙ)
end

function k₂ₙ(x, params::Vector{Float64})
    k₂(c₂ₙ(x, params)) * (ϵ₂ₙ ^ brugₙ)
end

function k₂ₙ(c₂ₙ::Float64)
    k₂(c₂ₙ) * (ϵ₂ₙ ^ brugₙ)
end

"""
Electrloyte conductivity in separator
"""
function k₂ₛ(params::Vector{Float64})
    k₂(c₂̄ₛ(params)) * (ϵ₂ₛ ^ brugₛ)
end

function k₂ₛ(x, params::Vector{Float64})
    k₂(c₂ₛ(x, params)) * (ϵ₂ₛ ^ brugₛ)
end

function k₂ₛ(c₂ₛ::Float64)
    k₂(c₂ₛ) * (ϵ₂ₛ ^ brugₛ)
end

"""
Electrloyte conductivity in positive electrode
"""
function k₂ₚ(params::Vector{Float64})
    k₂(c₂̄ₚ(params)) * (ϵ₂ₚ ^ brugₚ)
end

function k₂ₚ(x, params::Vector{Float64})
    k₂(c₂ₚ(x, params)) * (ϵ₂ₚ ^ brugₚ)
end

function k₂ₚ(c₂ₚ::Float64)
    k₂(c₂ₚ) * (ϵ₂ₚ ^ brugₚ)
end

File: scripts/new_ashwini/model/conductivity.m
% Include electrolyte_conc.m
run('electrolyte_conc.m')

% Effective electronic conductivity

% Effective electronic conductivity in negative electode
sigma_1_n = k_1_n * (epsilon_1_n ^ brug_n);

% Effective electronic conductivity in positive electode
sigma_1_p = k_1_p * (epsilon_1_p ^ brug_p);

% Ionic conductivity

% Electrloyte conductivity [Ref45 - 25]
function result = k_2(c_2)
    result = 1e-4*c_2*(-10.5 + 0.074*T - 6.69e-5*T^2 + 6.68e-4*c_2 - 1.78e-5*c_2*T + 2.8e-8*c_2*T^2 + 4.94e-7*c_2.^2 - 8.86e-10*T*c_2.^2).^2;
end

% Effective Electrloyte conductivity in negative electrode
function result = k_2_n(params)
    result = k_2(c_2_n_bar(params)) * (epsilon_2_n ^ brug_n);
end

function result = k_2_n(x, params)
    result = k_2(c_2_n(x, params)) * (epsilon_2_n ^ brug_n);
end

function result = k_2_n(c_2_n)
    result = k_2(c_2_n) * (epsilon_2_n ^ brug_n);
end

% Electrloyte conductivity in separator
function result = k_2_s(params)
    result = k_2(c_2_s_bar(params)) * (epsilon_2_s ^ brug_s);
end

function result = k_2_s(x, params)
    result = k_2(c_2_s(x, params)) * (epsilon_2_s ^ brug_s);
end

function result = k_2_s(c_2_s)
    result = k_2(c_2_s) * (epsilon_2_s ^ brug_s);
end

% Electrloyte conductivity in positive electrode
function result = k_2_p(params)
    result = k_2(c_2_p_bar(params)) * (epsilon_2_p ^ brug_p);
end

function result = k_2_p(x, params)
    result = k_2(c_2_p(x, params)) * (epsilon_2_p ^ brug_p);
end

function result = k_2_p(c_2_p)
    result = k_2(c_2_p) * (epsilon_2_p ^ brug_p);
end
